### PR TITLE
fix for unit(1, "strwidth", expression(x))

### PR DESCRIPTION
**Pull request by pmur002**
Here are the changes that I have made to your patch, based on a fork of your repo, so that there is just one patch to apply to R (rather than me trying to keep track of separate changes that I have made).  Sorry, I should have done this (forked your repo) from the start.

### CODE REVIEW

File: R/unit.R [hunk 1]
```diff
@@ -380,9 +381,9 @@ pSummary <- function(..., op) {
     x <- unclass(x)
     x[i] <- value
     if (simpleResult) {
-        `attributes<-`(x, attr)
+        attributes(x) <- attr
     } else {
-        `class<-`(x, "unit")
+        class(x) <- "unit"
     }
```
Comment (thomasp85) [inline R/unit.R]: What is the reason for this change? doesn't it prevent the object from being returned?

Comment (pmur002) [reply R/unit.R]: Oops.  Thanks for spotting that.  I missed returning 'x' at the end of the function.  Nice demonstration of why I should have been providing you with pull requests all along :)  
This change is now hopefully part of the pull request (?)

Comment (pmur002) [reply R/unit.R]: Are you happy to merge this so that I can just generate a single patch from the 'new-unit' branch to commit to r-devel ?

Comment (thomasp85) [reply R/unit.R]: I'll merge it into #1
